hrw.go

```diff
@@ -1,280 +1,124 @@
 // Package hrw implements Rendezvous hashing.
-// http://en.wikipedia.org/wiki/Rendezvous_hashing.
+// https://en.wikipedia.org/wiki/Rendezvous_hashing.
 package hrw
 
 import (
-	"encoding/binary"
-	"errors"
-	"math"
-	"reflect"
 	"sort"
 
 	"github.com/twmb/murmur3"
+	"golang.org/x/exp/constraints"
 )
 
-type (
-	// Hasher interface used by SortSliceByValue.
-	Hasher interface{ Hash() uint64 }
+// Hashable is something that can be hashed.
+type Hashable interface{ Hash() uint64 }
 
-	sorter struct {
-		l    int
-		less func(i, j int) bool
-		swap func(i, j int)
-	}
-)
-
-// Boundaries of valid normalized weights.
-const (
-	NormalizedMaxWeight = 1.0
-	NormalizedMinWeight = 0.0
-)
-
-func (s *sorter) Len() int           { return s.l }
-func (s *sorter) Less(i, j int) bool { return s.less(i, j) }
-func (s *sorter) Swap(i, j int)      { s.swap(i, j) }
-
-func distance(x uint64, y uint64) uint64 {
-	acc := x ^ y
-	// here used mmh3 64 bit finalizer
-	// https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L81
-	acc ^= acc >> 33
-	acc = acc * 0xff51afd7ed558ccd
-	acc ^= acc >> 33
-	acc = acc * 0xc4ceb9fe1a85ec53
-	acc ^= acc >> 33
-	return acc
+// Hash hashes the data in a fashion similar to the
+// [Hashable] interface. Can be used as a hash function
+// required for the HRW hashing algorithm.
+func Hash(data []byte) uint64 {
```

Comment:
Have you thought about `type HashableSlice []byte` that has `Hash() uint64` defined for it? Might be easier to use in some cases.

Reply:
do you mean to add an additional type and have both? or rework the version in this PR?

Reply:
well, i would say it's a part of its legacy to me. we had this func before, i kept it too. but agree that API is arguable and we can drop it.
but on the other side, it allows the provision of hashing func (that is not fixed strictly speaking by the HRW algorithm's basics). if someone wants to use our lib, it has to do some hashing for items being sorted and it will be harder with `type HashableSlice []byte` (e.g. we will have to marshall nodes, have some wrappers over them that return hashes)

Reply:
found direct usage: https://github.com/nspcc-dev/neofs-sdk-go/blob/1e86be9fa06bf4565ee824a66aba86f3060f2ce1/netmap/node_info.go#L227